components/org.wso2.carbon.crypto.provider/src/main/java/org/wso2/carbon/crypto/provider/SymmetricKeyInternalCryptoProvider.java

```diff
@@ -185,12 +248,44 @@ public byte[] encrypt(byte[] cleartext, String algorithm, String javaSecurityAPI
         return encrypt(cleartext, algorithm, javaSecurityAPIProvider);
     }
 
+    @Override
+    public byte[] encrypt(byte[] cleartext, String algorithm, String javaSecurityAPIProvider,
```

Comment:
we should be able to merge the common logic in above encrypt method and this method. Check for possibilities. That way, we can remove code duplications